cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -110,34 +90,112 @@ def behavior_for_path(behaviors, path)
   behaviors.detect do |behavior|
     paths = behavior[:path]
     next true unless paths
-    extensions, paths = normalize_paths(paths)
-    next true if extensions.any? && path.match(extensions_to_regex(extensions).first)
-    next true if paths.any?{|p| path.match path_to_regex(p) }
-    false
+    normalize_paths(paths).any?{|p| path.match p }
   end
 end
 
-
 # Generates the logic string for the specified behavior.
 def process_behavior(behavior, app, section)
-  if section == 'proxy'
-    process_proxy(behavior[:proxy] || app, app)
-  else
-    process_cookies(behavior[:cookies], section)
+  case section
+    when 'proxy'
+      process_proxy(behavior[:proxy] || app, app)
+    when 'vary'
+      process_vary(behavior)
+    when 'request'
+      process_request(behavior)
+    else
+      process_response(behavior)
   end
 end
 
-# Returns the cookie-filter string for a given 'cookies' behavior.
-def process_cookies(cookies, section)
-  if section == 'request'
-    return '# Allow all request cookies.' if cookies == 'all'
-    cookies = ['NO_CACHE'] if cookies == 'none'
-    "cookie.filter_except(\"#{cookies.join(',')}\");"
-  else
-    cookies == 'none' ?
-      'unset beresp.http.set-cookie;' :
-      '# Allow set-cookie responses.'
+# VCL string to create or update a Vary header field with the provided Vary header.
+def set_vary(header, resp)
+  # Matches a Vary header field delimiter.
+  sep = /\\s|,|^|$/
+  <<-VCL
+if (!#{resp}.http.Vary) {
+  set #{resp}.http.Vary = "#{header}";
+} elseif (#{resp}.http.Vary !~ "#{sep}#{header}#{sep}") {
+  set #{resp}.http.Vary = #{resp}.http.Vary + ", #{header}";
+}
+  VCL
+end
+
+# VCL string to set the appropriate Vary header fields based on the provided cache behavior.
+# Whitelisted headers are added to Vary directly.
+# Whitelisted cookies are added to Vary via their extracted X-COOKIE headers.
+def process_vary(behavior)
+  out = ''
+  behavior[:headers].each do |header|
+    out << set_vary(header, 'beresp')
+  end
+  case behavior[:cookies]
+    when 'all'
+      out << set_vary('Cookie', 'beresp')
+    when 'none'
+    else
+      behavior[:cookies].each do |cookie|
+      out << set_vary("X-COOKIE-#{cookie}", 'beresp')
+    end
```

Comment:
Lint/BlockAlignment: `end` at 139, 4 is not aligned with `behavior[:cookies].each do |cookie|` at 137, 6.